testing/bitstream/test_bits_utils_c.cpp

```diff
@@ -180,6 +180,55 @@ TEST_P(bits_utils_c_p, int_err1) {
   EXPECT_EQ(value, 0);
 }
 
+/**
+ * @test       double_err0
+ * @brief      Test: opae_bitstream_get_json_double
+ * @details    If the given name doesn't exist,<br>
+ *             the fn returns FPGA_EXCEPTION.<br>
+ */
+TEST_P(bits_utils_c_p, double_err0) {
+    const char *mdata =
+    R"mdata({
+"a": 3.14
+})mdata";
+  json_object *root;
+  double value = 0.0;
+
+  root = parse(mdata);
+  ASSERT_NE(root, nullptr);
+
+  EXPECT_EQ(opae_bitstream_get_json_double(root,
+					   "b",
+					   &value),
+            FPGA_EXCEPTION);
+  EXPECT_EQ(value, 0.0);
+}
+
+/**
+ * @test       double_err1
+ * @brief      Test: opae_bitstream_get_json_double
+ * @details    If the given name exists,<br>
+ *             but isn't of type double,<br>
+ *             the fn returns FPGA_EXCEPTION.<br>
+ */
+TEST_P(bits_utils_c_p, double_err1) {
+    const char *mdata =
+    R"mdata({
+"a": "str"
+})mdata";
+  json_object *root;
+  double value = 0.0;
+
+  root = parse(mdata);
+  ASSERT_NE(root, nullptr);
+
+  EXPECT_EQ(opae_bitstream_get_json_double(root,
+					   "a",
+					   &value),
+            FPGA_EXCEPTION);
+  EXPECT_EQ(value, 0.0);
+}
+
```

Comment:
Can we add a test for parsing something like:
```JSON
{
  "a": 42
}
```

I wonder if it is even necessary to call `opae_bitstream_get_json_int` if the get double fn fails.

Reply:
It appears to be.  Makes sense, because there is a separate json_type_X for double, ie both json_type_int and json_type_double exist.